platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -249,6 +269,15 @@ public void setupProjectLayout() {
             .ifPresent(link -> getLogger().lifecycle(link));
     }
 
+    private UserInputHandler getEffectiveInputHandler() {
+        Property<Boolean> noDialog = getNoDialog();
+        if (noDialog.isPresent() && noDialog.get()) {
```

Comment:
Instead of checking presence here, can you set a convention in `BuildInitPlugin` for `noDialog`?  This is what we did with `insecureProtocol`.

Reply:
Fixed